### PR TITLE
[GH-1345] Document the workload `Source` and Implementations

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadinstitute.atlassian.net/browse/GH-1345
Implementations include:
- `Terra DataRepo` Source
- `TDR Snapshots` Source

A good way to view this is to build and serve the docs locally:
```bash
make docs
. derived/.venv/docs/bin/activate && ( cd derived/docs; python3 -m mkdocs serve )
```
And go to http://127.0.0.1:8000/source.html

### CODE REVIEW

File: docs/md/source.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+determine which rows are new and/or have been updated and therefore need
+reprocessing. It should be a `Timestamp`, but `DateTime` is accepted too.
+You must ensure that the `Timestamp` or `DateTime` column uses Universal 
+Coordinated Time (UTC).
+
+!!! note
+    Using a `Timestamp` will increase the likelihood of workflow-launcher
+    detecting and scheduling new rows in real-time due to greater precision.
+    Using `DateTime` may cause workflow-launcher to miss the row at first 
+    (though it will be picked up later).
```
Comment (ehigham) [inline docs/md/source.md:77]: It wont - we should implement this!

Comment (ehigham) [reply docs/md/source.md:77]: https://broadinstitute.atlassian.net/browse/GH-1397

Comment (rexwangcc) [reply docs/md/source.md:77]: Agreed! We should defo implement this… Thanks for creating the ticket!

**Review by rexwangcc — APPROVED** [review]
Nice article! And maybe later we could put developers doc and user-facing guide into separate files. 

I guess it's intentional to not add `source.md` to the `mkdocs.yml` for now?

**Comment by ehigham** [comment]
> Nice article! And maybe later we could put developers doc and user-facing guide into separate files.
> 
> I guess it's intentional to not add `source.md` to the `mkdocs.yml` for now?

Yep - not sure where they'll go just yet. I think we'll combine them once #430 gets merged.

**Review by rhiananthony — APPROVED** [review]

**Pull request merged**